app/contexts/ThemeProvider.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import { ColorSchemeName, useColorScheme } from 'react-native'
+import { Logging } from '../api'
+import { ThemePersistence } from '../api/wallet/theme_storage'
+
+interface ThemeLoader {
+  theme: NonNullable<ColorSchemeName>
+  isThemeLoaded: boolean
+}
+
+export function useTheme (): ThemeLoader {
+  const colorScheme = useColorScheme()
+  const [isThemeLoaded, setIsThemeLoaded] = useState<boolean>(false)
+  const [theme, setTheme] = useState<NonNullable<ColorSchemeName>>('light')
+
+  useEffect(() => {
+    ThemePersistence.get().then((t) => {
```

Comment:
Maybe better if use just use system provided theme instead of having any overriding mechanism.

Similar to Language, always use system default.

Reply:
It always use system default. The overwriting only happens when user explicitly update it from settings page. So when they re-open the app, it should keep the change that they added.

Same for Language, it should be system default but we should still allow users to select and replace the default loaded if they want.